code/modules/admin/admin.dm

```diff
@@ -4483,7 +4483,7 @@ var/global/noir = 0
 
 	. = matches
 
-/proc/get_one_match(var/object, var/base = /atom, use_concrete_types=TRUE, only_admin_spawnable=TRUE)
+/proc/get_one_match(var/object, var/base = /atom, use_concrete_types=TRUE, only_admin_spawnable=TRUE, cmp_proc=/proc/cmp_text_asc)
```

Comment:
Imo default should be `null` to:
- keep the current behaviour
- give the caller the option not to sort. Sorting can be slow and sometimes thes get_on_match calls show an absolute crapton of stuff when you for example accidentally or intentionally enter something like "obj"

Reply:
if there was some way to filter this based off of the base type length or something that would be cool

Reply:
it's null for now, I guess we could add another arg, `sort_limit` that when set will sort any list below the length specificed with some sane default value, and otherwise won't bother

Reply:
i like that tbh